pkg/cmd/server/origin/run_components.go

```diff
@@ -265,9 +268,15 @@ func (c *MasterConfig) RunDeploymentController() {
 	if err != nil {
 		glog.Fatalf("Unable to initialize deployment controller: %v", err)
 	}
+
+	port, err := kubeserver.GetKubernetesMasterPort(c.Options)
```

Comment:
I still don't think this is right. The service port is not necessarily the same as the masters port